packages/react-server-dom-webpack/src/ReactFlightClientConfigBundlerWebpackBrowser.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+const chunkMap: Map<string, string> = new Map();
+
+/**
+ * We patch the chunk filename function in webpack to insert our own resolution
+ * of chunks that come from Flight and may not be known to the webpack runtime
+ */
+const webpackGetChunkFilename = __webpack_require__.u;
+__webpack_require__.u = function (chunkId: string) {
+  const flightChunk = chunkMap.get(chunkId);
+  if (flightChunk !== undefined) {
+    return flightChunk;
+  }
+  return webpackGetChunkFilename(chunkId);
+};
+
+export function loadChunk(chunkId: string, filename: string): Promise<mixed> {
+  chunkMap.set(chunkId, filename);
+  return __webpack_chunk_load__(chunkId);
+}
```

Comment:
@sebmarkbage why did we think we needed chunkId still on the client? With this implementation I can just send the filenames. I believe if there is a dynamic import for module Foo and module Foo is also in a client component chunk the dynamic import gets it's own chunk I think. webpack tries to load the dynamic chunk it'll have a different chunkId than the chunk used for client components. This is unfortunate b/c the code might already be on the client and if there were module state it wouldn't be shared across both instances. However if this is just how the bundling has to be there is no sense in serializing the chunkIds for client component imports because we can just look up the chunk in chunkMap using the filename rather than the chunkId (by passing the filename into `__webpack_chunk_load__`

Reply:
I think the answer I don't think Webpack does duplicate the modules. If it does, then we need to fix that. Maybe there's a bug in Next.js but I wouldn't expect the prototype implementation do that. Can you confirm?

Reply:
I confirmed you are correct